### ISSUE TITLE
Which header to use?

### ISSUE THREAD

**Issue by CaptainChicky** [cscape/vitalsource-dl#4]
There are a lot of headers in the page, so which one do I use?

**Comment by 99bobster99** [comment]
I have the same issue, not knowing which header to use. Did you ever resolve this issue?

**Comment by CaptainChicky** [comment]
Unfortunately not. I have tried multiple methods, but were unable to get any of them to work. The best bet would be for me to write a python script that uses inspect element to get the images of the textbook and then one can merge them into a PDF, but tbh, I'm too lazy.

**Comment by 99bobster99** [comment]
Yeah, I think a lot of these scripts for Vitalsource got nuked with some of their latest updates. I tried a couple of scripts as well, they all seem to get my web access "unauthorized". :(  I'm lazy too, I will just use their offline app, much easier! :)

**Comment by CaptainChicky** [comment]
Aight I think I should have the time to write stuff now.  🤔 I can probably just do a script where it auto clicks each page for you. Shouldn't be too hard.

**Comment by 99bobster99** [comment]
That would be great, please let me know when you have something to try. Good luck! :)

**Comment by ciroiriarte** [comment]
more than a variable, probably it would be more practical to export the cookies from the browser as a file and have the script loading them from the file instead of depending on a string variable.

**Comment by CaptainChicky** [comment]
oh dang its been 2 years and funnily (or sadly?) enough this issue is still open rippp


> That would be great, please let me know when you have something to try. Good luck! :)

Anyhow, hey @99bobster99 I know this is extremely late, but I feel like I do need to give a reply for closure. I'll admit I kinda forgot about this issue post here, and what I eventually did was a brute force technique with VBScript. I loaded the vitalsource textbook on my browser such that iirc it was something like 
`https://blah blah/blah/blah/<page number>`, and I used VBScript to go through an inspect element sequence and save the image. Repeat this 1000+ times and you can save a textbook this way. 

It was extremely cumbersome and I can't seem to find the vbs file now that it's been two years. But basically, you're just going through a while true loop click (whether its a click on a specific coordinate point on the computer screen or a keypress) cycle with an inner loop to loop through the page numbers. I had to run it manually for each page number increase in digit (so from like 99 -> 100 I had to redo the inner loop iirc)

I think you probably can quite easily replicate this but again, I really dont recommend this this is kind of dumb lol. Regardless, my classes use webassign now so I don't have any access to test with vitalsource anymore. My apologies and hopefully you had better luck than me.

**Comment by calvinhobbes23** [comment]
Can you help. I got 5 books that are going to expire in a month.

**Comment by CaptainChicky** [comment]
> Can you help. I got 5 books that are going to expire in a month.

just grab the book off of libgen tbh lol

**Comment by 99bobster99** [comment]
Hey Captain Chicky, thank you so much for remembering to get back to me on my request! I received an email alert on calvinhobbes23's request and then saw your answer to my question. I'm glad you did, since I need to get some more books for both my University kids. You rock sir, thank you so much!!

**Comment by calvinhobbes23** [comment]
Unfortunately Libgen doesnt have them. Those are AWS Training student guide material I got when I took AWS training. They are going to expire in a couple months.

**Comment by CaptainChicky** [comment]
@99bobster99 glad to be at least somewhat of help 😅, though if the vbs thing isn't appealing or is cumbersome you can try to give [this](https://github.com/Cyberes/vitalsource2pdf) a try (after fixing it)

@calvinhobbes23 you can try that as well^ or maybe try looking for the books on archive.org

If it comes to naught, honestly just spend like a few hours of a day manually saving each page until you have the book. I currently don't have any classes using vitalsouce so I don't have any way of developing a downloader tool, but if i ever do in the future I'll see what i can do then kek

**Comment by ciroiriarte** [comment]
> Unfortunately Libgen doesnt have them. Those are AWS Training student guide material I got when I took AWS training. They are going to expire in a couple months.

Share session cookies & book IDs, I'll give it a go.

Note: couldn't find your email :)
